src/index.ts

```diff
@@ -208,12 +209,27 @@ function startApp(appSchema, path: string) {
   })
 
   app.use("/batch", bodyParser.json(), graphqlBatchHTTPWrapper(graphqlServer))
-  app.use(graphqlServer)
+  app.use(graphqlUploadMiddleware, graphqlServer)
```

Comment:
**question** Is there a difference between having these in the same vs different `app.use` invocations? Also, would we consider having an explicit feature flag for this middleware instead of relying on "not production"?

```typescript
const { ENABLE_GRAPHQL_UPLOAD } = config

if (ENABLE_GRAPHQL_UPLOAD) {
  app.use(graphqlUploadExpress({
    maxFileSize: GRAPHQL_UPLOAD_MAX_FILE_SIZE_IN_BYTES,
    maxFiles: GRAPHQL_UPLOAD_MAX_FILES,
  }))
}

app.use(graphqlServer)
```

Reply:
> would we consider having an explicit feature flag for this middleware instead of relying on "not production"?

Do you want to read this value from `process.env`?

Reply:
> Is there a difference between having these in the same vs different app.use invocations?

If I understand you correctly, there is no difference. We needed to perform some additional checks ([like this](https://github.com/artsy/metaphysics/blob/530d65e7dcbd44aa8087c2419451419d400a2233/src/lib/graphqlUpload.ts#L14-L16)) and for this reason I created `graphqlUploadMiddleware` for this

Reply:
Yep, our configuration setup relies on `process.env` so we can follow that pattern for a new `ENABLE_GRAPHQL_UPLOAD` flag ([example](https://github.com/artsy/force/blob/6724e12b80e75655d29a46f541ad364c5efb47c7/src/config.ts#L50-L57)).

Reply:
Added